x-pack/plugin/core/src/main/java/org/elasticsearch/xpack/core/security/authc/RealmConfig.java

```diff
@@ -30,7 +30,7 @@ public RealmConfig(RealmIdentifier identifier, Settings settings, Environment en
         this.env = env;
         this.threadContext = threadContext;
         this.enabled = getSetting(RealmSettings.ENABLED_SETTING);
-        if (false == hasSetting(RealmSettings.ORDER_SETTING.apply(type()))) {
+        if (enabled && false == hasSetting(RealmSettings.ORDER_SETTING.apply(type()))) {
```

Comment:
Nit: Okay, but I would slightly prefer we move the `IllegalArgumentException` validation in `Realms#buildRealmConfigs` or `Realms#initRealms`, as I think it would be easier to follow together with all the others.
As an example, admins could still get the "duplicated realms" exception if there are two native realms but only one is enabled (which is a similar situation to missing the order on a disabled realm, ie whether disabled realms obey the validations).
But this is just a nit, if you think it requires to much code moving around, that's Okay.

Reply:
I'd prefer to not change it here mainly because this PR is 8.0 only, while change of the validation location is something that should be done for both 7.x and 8.0. So a separate PR is better suited. 

PS: we had the [same](https://github.com/elastic/elasticsearch/pull/51195#discussion_r370224914) [conversation](https://github.com/elastic/elasticsearch/pull/51195#discussion_r370228928) about the location of this check when it was originally introduced. Your showed a great consistency here. I think that itself warrants another look at the choice. I'll think about it more carefully after the current PR. Thanks!